File: VB/Controls/LogViewerVB.WinForms/Extensions/ControlsExtension.vb
Imports System.Reflection
Imports System.Runtime.CompilerServices

Public Module ControlsExtension

    <Extension>
    Public Sub AddControl(panel As Panel, control As Control)

        panel.Controls.Add(control)
        control.Dock = DockStyle.Fill
        control.BringToFront()

    End Sub

    ' ref: https://stackoverflow.com/a/42389596
    <Extension>
    Public Sub SetDoubleBuffered(control As Control, Optional doubleBuffered As Boolean = True)
        control.[GetType]().GetProperty("DoubleBuffered", BindingFlags.Instance Or BindingFlags.NonPublic)?.SetValue(control, doubleBuffered, Nothing)
    End Sub

End Module

File: CSharp/Controls/LogViewer.WinForms/Extensions/ControlsExtension.cs
namespace LogViewer.WinForms;

public static class ControlsExtension
{
    public static void AddControl(this Panel panel, Control control)
    {
        panel.Controls.Add(control);
        control.Dock = DockStyle.Fill;
        control.BringToFront();
    }

    // ref: https://stackoverflow.com/a/42389596
    public static void SetDoubleBuffered(this Control control, bool doubleBuffered = true)
    {
        control
            .GetType()
            .GetProperty("DoubleBuffered", System.Reflection.BindingFlags.Instance | System.Reflection.BindingFlags.NonPublic)
            ?.SetValue(control, doubleBuffered, null);
    }
}